### PR TITLE
Adding library id and target graphs for log messages

### PR DESCRIPTION
**Pull request by emgarten**
Cleaning up warning and error logging, adding library ids and target graphs for all relevant messages.

Fixes https://github.com/NuGet/Home/issues/5405

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs [hunk 1]
```diff
@@ -483,7 +484,7 @@ private static Task LogDowngradeWarningsAsync(IEnumerable<RestoreTargetGraph> gr
             return logger.LogMessagesAsync(mergedMessages);
         }
 
-        private static IList<CompatibilityCheckResult> VerifyCompatibility(
+        private static async Task<IList<CompatibilityCheckResult>> VerifyCompatibility(
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs]: `VerifyCompatibilityAsync`

**Review by alpaix — APPROVED** [review]
Kudos on great tests!

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs (hunk 1 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs]: Good catch, I'll rename this

**Review by jainaashish — APPROVED** [review]
LGTM

**Comment by emgarten** [comment]
Merged with NuGet/NuGet.Client@f543d5f592729485b1ac3dad37388537e1dcec96

**Pull request closed without merging**